crates/fluvio-protocol-derive/src/ast/prop.rs

```diff
@@ -151,76 +143,322 @@ impl UnnamedProp {
 }
-
-pub fn validate_versions(min: i16, max: Option<i16>, field: Option<&str>) -> Option<String> {
-    match (max, field) {
-        // Print name in named fields
-        (Some(max), Some(field)) if min > max => Some(format!(
-            "On {field}, max version({max}) is less than min({min})."
-        )),
-        // No name to print in unnamed fields
-        (Some(max), None) if min > max => {
-            Some(format!("Max version({max}) is less than min({min})."))
-        }
-        (None, Some(field)) if min < 0 => {
-            Some(format!("On {field} min version({min}) must be positive."))
+/// Convert the values to TokenStream which will be ready to use variable value
+///
+/// # Example
+/// ````ignore
+/// // Function as a literal
+/// fn test() -> i16 { 1 }
+/// #[fluvio(min_version = "test()")]
+/// ````
+/// To use the value from the test() function:
+/// ````ignore
+/// let func_value = prop_attrs_type_value(prop_attr_type, None)
+/// ````
+/// To set a specific type you can do this:
+/// ````ignore
+/// let ident_type = Ident::new("u8", Span::call_site());
+/// let func_value = prop_attrs_type_value(prop_attr_type, Some(&ident_type))
+/// ````
+///
+pub fn prop_attrs_type_value(
+    attrs_type: Option<&PropAttrsType>,
+    ident_type: Option<&Ident>,
+) -> TokenStream {
+    if let Some(attr) = attrs_type {
+        match &attr {
+            PropAttrsType::Lit(data) => parse_quote!(#data),
+            PropAttrsType::Fn(data) => TokenStream::from_str(&format!("{}()", data)).unwrap(),
+            PropAttrsType::Int(data) => {
+                if let Some(itype) = ident_type {
+                    TokenStream::from_str(&format!("{}_{}", data, itype)).unwrap()
+                } else {
+                    // By default it's i16, because most places use it
+                    TokenStream::from_str(&format!("{}_i16", data)).unwrap()
+                }
+            }
+            PropAttrsType::None => parse_quote!(0_i16),
         }
-        (None, None) if min < 0 => Some(format!("Min version({min}) must be positive.")),
-        _ => None,
+    } else {
+        parse_quote!(0_i16)
     }
 }
-
+/// A type that will handle the values passed in properties
+/// and convert them later on to TokenStream.
+///
+/// Using this type allows you to pass values multiple ways:
+/// # Example
+///
+/// ```ignore
+/// // Constant as a path
+/// const TEST: i16 = 1;
+/// #[fluvio(min_version = TEST)]
+/// ```
+///
+/// ```ignore
+/// // Constant as a literal
+/// const TEST: i16 = 1;
+/// #[fluvio(min_version = "TEST")]
+/// ```
+///
+/// ```ignore
+/// // Function as a literal
+/// fn test() -> i16 { 1 }
+/// #[fluvio(min_version = "test()")]
+/// ```
+///
+/// ```ignore
+/// // Int
+/// #[fluvio(min_version = 1)]
+/// ```
+///
+/// None has a default Int value of 0 which is set in prop_attrs_type_value
+#[derive(Default, Clone)]
+pub enum PropAttrsType {
+    Lit(Ident),
+    Fn(Ident),
+    Int(i16),
+    #[default]
+    None,
```

Comment:
Why do we need `None` variant? Property attributes can't be without type or with undefined type, right?

Reply:
I thought it might be cleaner to do it like that so all the defaults are set in the `prop_attrs_type_value`, but it might be cleaner to do it without the `None` and just set the default as `Int(0)` so will change it with the new changes.